crates/sui-json-rpc/src/coin_api.rs

```diff
@@ -550,7 +550,7 @@ mod tests {
         let previous_transaction = TransactionDigest::from(arr);
         let object = Object::new_move(
             MoveObject::new_coin(
-                coin_type_string.parse::<StructTag>().unwrap().into(),
+                coin_type_string.parse::<TypeTag>().unwrap(),
```

Comment:
Why switch from `TypeTag` instead of `StructTag`? A coin's marker type must be a `StructTag` (i.e. it can't be a primitive).

Reply:
This is primarily to be consistent with the definition of Coin in MoveObjectTag:
https://github.com/MystenLabs/sui/blob/main/crates/sui-types/src/base_types.rs#L180
which I think we can no longer change.
It was probably a oversight back then?